scalafmt-sbt/src/main/scala/org/scalafmt/sbt/ScalafmtPlugin.scala

```diff
@@ -1,25 +1,169 @@
 package org.scalafmt.sbt
-import scala.meta.internal.tokenizers.PlatformTokenizerCache
-import sbt._
-import Keys._
+
+import org.scalafmt.{Formatted, Scalafmt}
+import org.scalafmt.config.{Config, ScalafmtConfig, ScalafmtRunner}
+import sbt.Keys._
+import sbt.{Def, _}
+import complete.DefaultParsers._
+import sbt.util.Logger
+
+import scala.util.{Failure, Success, Try}
 
 object ScalafmtPlugin extends AutoPlugin {
   override def trigger: PluginTrigger = allRequirements
+
   object autoImport {
-    val scalafmt: Command =
-      Command.args("scalafmt", "run the scalafmt command line interface.") {
-        case (state, args) =>
-          org.scalafmt.cli.Cli.main("--non-interactive" +: args.toArray)
-          PlatformTokenizerCache.megaCache.clear()
-          state
+    lazy val scalafmt = TaskKey[Unit]("scalafmt", "Format Scala sources")
+    lazy val scalafmtCheck = TaskKey[Boolean](
+      "scalafmt-check",
+      "Check that Scala sources is formatted properly")
+    lazy val scalafmtOnCompile =
+      SettingKey[Boolean]("scalafmt-on-compile", "Format source when compiling")
+    lazy val scalafmtConfig =
+      TaskKey[File]("scalafmt-config", "Scalafmtter config file")
+    lazy val scalafmtSbt = TaskKey[Unit]("scalafmt-sbt", "Format SBT sources")
+    lazy val scalafmtSbtCheck = TaskKey[Boolean](
+      "scalafmtSbtCheck",
+      "Check that SBT sources is formatted properly")
+    lazy val scalafmtOnly =
+      InputKey[Unit]("scalafmt-only", "Format only given file")
+  }
+  import autoImport._
+
+  private lazy val scalaConfig =
```

Comment:
Call this `scalafmtConfigParsed` instead?

Reply:
this means that if the config file changes, sbt needs to be restarted. It would be more idiomatic to assign this to a Task that can be cached based on the file.

Reply:
`scalafmtConfig` is a `TaskKey[File]`, and this is calling map on it, which will return an `Def.Initialize[Task[Configured[ScalafmtConfig]]]` (if I understand correctly) so it's ok.